Bump gradle-version, dependencies, support JDK 11 (#1118)
mapbox/mapbox-maps-android: DEVELOPING.md

[`sdkmanager`](https://developer.android.com/studio/command-line/sdkmanager) command line or
through the GUI interface to make sure the extra requirements to build Android are met.

> Note: We use Android gradle plugin v7.0+ to build the project, which requires JDK 11.

Reviewer: ```suggestion
Note: Android Gradle Plugin v7.0+ is used to build the project. That requires to have JDK 11 or higher.
```

> when building project, make sure you have JDK 11 installed and set it as JAVA_HOME variable.

Reviewer: I'd drop this line at all and use my suggestion above.
